python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -215,6 +221,90 @@ def execute(self, settings: CLISettings) -> None:
                             fetched_data,
                         )
 
+    def validate(
+        self,
+        settings: CLISettings,
+        hashes_grouped_by_prefix: t.Dict[t.Optional[str], t.Set[str]],
+        signal_types: t.List[t.Type[SignalType]],
+    ) -> t.Dict[t.Type[SignalType], t.Set[str]]:
+        """
+        Takes the hashes grouped by optional string prefix, performs all required validations.
+        Command line arguments, SignalType, hashes, and ambiguous SignalTypes are validated.
+        Returns a dict of validated SignalType, and validated hashes of each SignalType.
```

Comment:
Thanks for adding a docstring!

One detail (similar to above comment) is that this validates when the input is hashes